### PR TITLE
Add support for path-regex annotation in Ingress Master-Minion

### PR DESCRIPTION
**Pull request by jjngx**
### Proposed changes

This PR adds support for setting up nginx.org/path-regex annotations
for Ingress and Ingress Master - Minion setup.

When this PR is merged, users will be able:

- Configure `nginx.org/path-regex` annotations on Minions and Master Ingress independently.
- Configure `nginx.org/path-regex` annotations (regex modifiers) `case_sensitive`, `case_insensitive` and `exact` independently on each minion or master

Annotations set on Minions take higher priority over annotations set on Master.
PR with documentation updates will follow once this one is merged with the main branch.

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [X] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [X] I have rebased my branch onto main
- [X] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: internal/configs/version1/template_helper.go [hunk 1]
```diff
@@ -14,18 +14,42 @@ func trim(s string) string {
...
 
-// makeLocationPath takes a string representing a location path
-// and a map representing Ingress annotations.
+// makeLocationPath takes location and Ingress annotations and returns
+// modified location path with added regex modifier or the original path
+// if no path-regex annotation is present in ingressAnnotations
+// or in Location's Ingress.
+//
+// Annotation 'path-regex' set on the Minion takes higher priority comparing
+// to the annotation set on the master (in Master-Minion Ingress setup).
```
Comment (shaun-nx) [inline internal/configs/version1/template_helper.go:23]:
```suggestion
// Annotation 'path-regex' set on a Minion Ingress will take priority over
// the annotation set on the master (in Master-Minion Ingress setup).
```

Comment (shaun-nx) [reply internal/configs/version1/template_helper.go:23]: Slight re-wording. The phrase `takes higher priority` makes me think that this is a priority that can be adjusted.

**Review by shaun-nx — APPROVED** [review]
Great stuff Jakub. A lot of this new testing code is going to make template testing much easier going forward 🚀

File: internal/configs/version1/template_helper.go (hunk 1 above)
Comment (jjngx) [reply internal/configs/version1/template_helper.go]: Applied change in the new commit, thanks @shaun-nx !

**Review by haywoodsh — APPROVED** [review]

**Pull request merged**